pkg/transform/kubernetes/kubernetes.go

```diff
@@ -38,6 +38,13 @@ func createServices(app *spec.App) []runtime.Object {
 		if len(svc.Spec.Selector) == 0 {
 			svc.Spec.Selector = app.Labels
 		}
+		if s.Name == "" {
```

Comment:
Or we could at least put everything in one place. Something like `func populateNames(app *spec.App)` then we will have all that in one place. I'm afraid that this will make it unreadable really soon.